tests/rkt_fetch_test.go

```diff
@@ -168,3 +174,180 @@ func testFetchNoStore(t *testing.T, args string, image string, finalURL string)
 	}
 	child.Wait()
 }
+
+func TestResumedFetch(t *testing.T) {
+	image := "rkt-inspect-implicit-fetch.aci"
+	imagePath := patchTestACI(image, "--exec=/inspect")
+
+	defer os.Remove(imagePath)
+
+	kill := make(chan struct{})
+	reportkill := make(chan struct{})
+
+	server := httptest.NewServer(testInterruptingServerHandler(t, imagePath, kill, reportkill, true, false))
+	defer server.Close()
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	cmd := fmt.Sprintf("%s --no-store --insecure-skip-verify fetch %s", ctx.Cmd(), server.URL)
+	child := spawnOrFail(t, cmd)
+	<-kill
+	err := child.Close()
+	if err != nil {
+		panic(err)
+	}
+	reportkill <- struct{}{}
+
+	// rkt has fetched the first half of the image
+	// If it fetches the first half again these channels will be written to.
+	// Closing them to make sure this doesn't happen.
+	close(kill)
+	close(reportkill)
+
+	child = spawnOrFail(t, cmd)
+	err = child.Wait()
+	if err != nil {
+		t.Errorf("rkt didn't exit cleanly: %v", err)
+	}
+}
+
+func TestResumedFetchLastModUnsupported(t *testing.T) {
+	image := "rkt-inspect-implicit-fetch.aci"
+	imagePath := patchTestACI(image, "--exec=/inspect")
+
+	defer os.Remove(imagePath)
+
+	kill := make(chan struct{})
+	reportkill := make(chan struct{})
+
+	server := httptest.NewServer(testInterruptingServerHandler(t, imagePath, kill, reportkill, false, false))
+	defer server.Close()
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	// rkt should fetch from the start of the image both times in this loop,
+	// since the server won't fill out the Last-Modified header. If rkt doesn't
+	// start the second fetch from the start of the image, the `<-kill` line
+	// will wait indefinitely.
```

Comment:
Can we instead test this sequence:
- 1. download interrupted in the middle
- 2. complete download (and check the sha512 hash)

in order to check that rkt can succeed a full download in this scenario?